### ISSUE TITLE
multiple checks for a constant value

### ISSUE THREAD

**Issue by KotlinIsland** [DetachHead/basedpyright#103]
```py
import sys

if sys.platform == "win32":
    def f(): ...

if sys.platform == "win32":
    f()  # error: possibly undefined
```

As this funtion is defined within a condition that is equal to the usage, and the value is `Final`, it should not show an error.
